packages/react-router-dev/vite/plugin.ts

```diff
@@ -1448,7 +1448,7 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = () => {
           cacheDir: "node_modules/.vite-child-compiler",
           mode: viteConfig.mode,
           server: {
-            watch: viteConfig.command === "build" ? null : undefined,
+            watch: viteConfig.command === "build" ? null : viteConfig.server.watch,
```

Comment:
Just checking, but is `viteConfig.server` always defined?

Reply:
I double-checked this on my project. Removing the `server` section in `vite.config.ts` does not lead to errors.